.github/workflows/CI.yml

```
@@ -35,4 +32,23 @@ jobs:
       - name: Run build
-        run: uv build
\ No newline at end of file
+        run: uv build
+  pre-commit:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.11", "3.12"]
+    steps:
+    - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)